File: python/src/copilot_protocol/fixtures.py
from __future__ import annotations

import json
from pathlib import Path
from typing import Any


def load_fixture(path: str | Path) -> dict[str, Any]:
    return json.loads(Path(path).read_text(encoding="utf-8"))


def fixtures_dir() -> Path:
    return Path(__file__).resolve().parents[3] / "fixtures"

File: typescript/src/fixtures.ts
import fs from "node:fs";
import path from "node:path";
import { fileURLToPath } from "node:url";

const __dirname = path.dirname(fileURLToPath(import.meta.url));

export function fixturesDir(): string {
  return path.resolve(__dirname, "../../fixtures");
}

export function loadFixture(filePath: string): {
  name: string;
  direction: string;
  frames: Record<string, unknown>[];
} {
  return JSON.parse(fs.readFileSync(filePath, "utf-8"));
}
